paasta_tools/kubernetes_tools.py

```diff
@@ -1516,8 +1516,63 @@ def get_pod_volumes(
         if crypto_volume:
             pod_volumes.append(crypto_volume)
 
+        database_credentials_secrets_volumes = (
+            self.get_database_credentials_secrets_volumes()
+        )
+        if database_credentials_secrets_volumes:
+            # note an array is returned so we extend vs append
+            pod_volumes.extend(database_credentials_secrets_volumes)
+
         return pod_volumes
 
+    def get_database_credentials(self) -> List[str]:
+        database_credentials = self.config_dict.get("database_credentials", {})
+        return database_credentials
+
+    def get_database_credentials_secret_name(self, dstore="", credential="") -> str:
+        return _get_secret_name(
+            self.get_namespace(),
+            "database-credentials",
+            self.get_service(),
+            f"{dstore}-{credential}",
+        )
+
+    def get_database_credentials_volume_name(self, dstore="", credential="") -> str:
+        return limit_size_with_hash(
+            name=f"database-credentials-volume-{dstore}-{credential}-{self.get_sanitised_service_name()}".replace(
+                "_", "--"
+            )
+        )
+
+    def get_database_credentials_secrets_volumes(self) -> Optional[V1Volume]:
+        database_credentials = self.get_database_credentials()
+        if not database_credentials:
+            return None
+
+        result = []
+        for dstore in database_credentials.keys():
+            for credential in database_credentials[dstore]:
```

Comment:
nit: might be nice to use a bit of python magic here :)
```suggestion
        for dstore, credentials in database_credentials.items():
            for credential in credentials:
```